File: src/dataset/javascript/say_exercism-ThomasZumsteg/openAI/ChatGPT4_say_v3.js
const ones = ["", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
const teens = ["", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"];
const tens = ["", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"];
const powers = ["", " thousand", " million", " billion"];

function inEnglish(num) {
    if (num < 0 || num >= 1_000_000_000_000) throw new Error('Number must be between 0 and 999,999,999,999.');
    if (num === 0) return "zero";

    const digitGroups = [];
    while (num > 0) {
        digitGroups.push(num % 1000);
        num = Math.floor(num / 1000);
    }

    return digitGroups
        .map((group, power) => group > 0 ? englishPower(group) + powers[power] : "")
        .filter(Boolean)
        .reverse()
        .join(" ");
}

function englishPower(digits) {
    const [hundreds, tensDigit, onesDigit] = String(digits).padStart(3, '0').split('').map(Number);
    let english = hundreds ? `${ones[hundreds]} hundred` : "";

    if (tensDigit === 1 && onesDigit > 0) {
        english += (english ? " " : "") + teens[onesDigit];
    } else {
        if (tensDigit > 0) english += (english ? " " : "") + tens[tensDigit];
        if (onesDigit > 0) english += (english && tensDigit > 0 ? "-" : " ") + ones[onesDigit];
    }

    return english.trim();
}

module.exports = { inEnglish };

File: src/dataset/python/say_exercism-ThomasZumsteg/openAI/ChatGPT4_say_v3.py
NUMBERS = ['', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
TENS = ['', 'ten', 'twenty', 'thirty', 'forty', 'fifty', 'sixty', 'seventy', 'eighty', 'ninety']
TEENS = ['ten', 'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen', 'sixteen', 'seventeen', 'eighteen', 'nineteen']
POWERS = ['', ' thousand', ' million', ' billion']

def say(num):
    if not (0 <= num < 1_000_000_000_000):
        raise AttributeError("Number must be between 0 and one trillion")
    if num == 0:
        return 'zero'

    parts = []
    for p, power_name in enumerate(POWERS):
        num, power = divmod(num, 1000)
        if power > 0:
            part = say_power(power) + power_name
            parts.append(part)
        if num == 0:
            break

    return ' '.join(reversed(parts)).replace(' and ', ' ', 1) if len(parts) > 1 else parts[0]

def say_power(num):
    hundreds, remainder = divmod(num, 100)
    tens, ones = divmod(remainder, 10)
    parts = []

    if hundreds:
        parts.append(f"{NUMBERS[hundreds]} hundred")
    if tens == 1:
        parts.append(TEENS[ones])
    else:
        if tens:
            parts.append(TENS[tens])
        if ones:
            parts.append(NUMBERS[ones])

    return ' and '.join(parts) if hundreds and remainder else ' '.join(parts)